app/controllers/application_controller.rb

```diff
@@ -152,10 +152,14 @@ def setup_user_auth
     # have we identified the user?
     if @user
       # check if the user has been banned
-      if @user.blocks.active.exists?
-        # NOTE: need slightly more helpful message than this.
+      user_block =  @user.blocks.active.take
+      unless user_block.nil? 
         set_locale
-        report_error t("application.setup_user_auth.blocked"), :forbidden
+        if  @user.blocks.active.take.zero_hour?
```

Comment:
Too much chatting at the hack weekend and committed before I had actually finished changing the code.  Sigh.